### PR TITLE
[12.0][IMP] agreement_serviceprofile: Allow Product Variant for Serviceprofile

### PR DESCRIPTION
**Pull request by osi-scampbell**
Issue #546

### CODE REVIEW

**Comment by patrickrwilson** [comment]
@osi-scampbell When I go to add the variant to the service profile on an agreement, I think the field is missing from the form...
image

File: agreement_serviceprofile/views/agreement.xml [hunk 1]
```diff
@@ -12,14 +12,17 @@
...
                         </tree>
                         <form>
                             <group>
                                 <group>
                                     <field name="name"/>
                                 </group>
                                 <group>
-                                    <field name="product_id"/>
+                                    <field name="use_product_variant"/>
+                                    <field name="product_variant_id" attrs="{'invisible': [('use_product_variant', '=', False)], 'required': [('use_product_variant','!=', False)]}"/>
```
Comment (bodedra) [inline agreement_serviceprofile/views/agreement.xml]:
```suggestion
                                    <field name="product_variant_id" context="{'default_is_serviceprofile': True, 'default_type': 'service'}" attrs="{'invisible': [('use_product_variant', '=', False)], 'required': [('use_product_variant','!=', False)]}"/>
```

File: agreement_serviceprofile/views/agreement.xml [hunk 2]
```diff
@@ -12,14 +12,17 @@
...
+                                    <field name="product_id" attrs="{'invisible': [('use_product_variant', '!=', False)], 'required': [('use_product_variant','=', False)]}"/>
```
Comment (bodedra) [inline agreement_serviceprofile/views/agreement.xml]:
```suggestion
                                    <field name="product_id" context="{'default_is_serviceprofile': True, 'default_type': 'service'}" attrs="{'invisible': [('use_product_variant', '!=', False)], 'required': [('use_product_variant','=', False)]}"/>
```

**Review by bodedra — CHANGES_REQUESTED** [review]

File: agreement_serviceprofile/views/agreement_serviceprofile.xml [hunk 3]
```diff
@@ -47,7 +48,9 @@
                             <field name="agreement_id"/>
                         </group>
                         <group>
-                            <field name="product_id"/>
+                            <field name="product_id" attrs="{'invisible': [('use_product_variant', '!=', False)], 'required': [('use_product_variant','=', False)]}"/>
```
Comment (bodedra) [inline agreement_serviceprofile/views/agreement_serviceprofile.xml]:
```suggestion
                            <field name="product_id" context="{'default_is_serviceprofile': True, 'default_type': 'service'}" attrs="{'invisible': [('use_product_variant', '!=', False)], 'required': [('use_product_variant','=', False)]}"/>
```

File: agreement_serviceprofile/views/agreement_serviceprofile.xml [hunk 4]
```diff
@@ -47,7 +48,9 @@
...
+                            <field name="product_variant_id" attrs="{'invisible': [('use_product_variant', '=', False)], 'required': [('use_product_variant','!=', False)]}"/>
```
Comment (bodedra) [inline agreement_serviceprofile/views/agreement_serviceprofile.xml]:
```suggestion
                            <field name="product_variant_id" context="{'default_is_serviceprofile': True, 'default_type': 'service'}" attrs="{'invisible': [('use_product_variant', '=', False)], 'required': [('use_product_variant','!=', False)]}"/>
```

**Review by bodedra — CHANGES_REQUESTED** [review]

**Review by bodedra — APPROVED** [review]
Functional reviewed. :+1:

**Review by patrickrwilson — APPROVED** [review]
LGTM

**Pull request merged**